crates/environ/src/component/translate.rs

```diff
@@ -622,8 +622,29 @@ impl<'a, 'data> Translator<'a, 'data> {
                     self.result
                         .initializers
                         .push(LocalInitializer::Export(item));
-                    if let ComponentItem::Type(ty) = item {
-                        self.types.push_component_typedef(ty);
+
+                    // Exports create a new index, so push the item onto the
+                    // appropriate list.
+                    match item {
+                        ComponentItem::Func(idx) => {
+                            self.result
+                                .component_funcs
+                                .push(self.result.component_funcs[idx]);
+                        }
+                        ComponentItem::Module(_idx) => {
+                            // ???
```

Comment:
@alexcrichton What should we do here?

Reply:
I think this entry can be ignored since we don't track the type information about modules here in this pass. For example translation of `Payload::ModuleSection` doesn't do anything to various index spaces.